src/execution_plan/ops/op_sort.c

```diff
@@ -12,56 +12,60 @@
 #include "../../util/rmalloc.h"
 #include "../../query_ctx.h"
 
-/* Forward declarations. */
+// forward declarations
 static OpResult SortInit(OpBase *opBase);
 static Record SortConsume(OpBase *opBase);
 static OpResult SortReset(OpBase *opBase);
 static OpBase *SortClone(const ExecutionPlan *plan, const OpBase *opBase);
 static void SortFree(OpBase *opBase);
 
-// Heapsort function to compare two records on a subset of fields.
-// Return value similar to strcmp.
-static int _record_compare(Record a, Record b, const OpSort *op) {
+// function to compare two records on a subset of fields
+// return value similar to strcmp
+static int _record_cmp
+(
+	Record a,
+	Record b,
+	OpSort *op
+) {
 	uint comparison_count = array_len(op->record_offsets);
 	for(uint i = 0; i < comparison_count; i++) {
 		SIValue aVal = Record_Get(a, op->record_offsets[i]);
 		SIValue bVal = Record_Get(b, op->record_offsets[i]);
 		int rel = SIValue_Compare(aVal, bVal, NULL);
-		if(rel == 0) continue;      // Elements are equal; try next ORDER BY element.
-		rel *= op->directions[i];   // Flip value for descending order.
+		if(rel == 0) continue; // elements are equal; try next ORDER BY element
+		rel *= op->directions[i]; // flip value for descending order
 		return rel;
 	}
-	return 0;
-}
 
-// Quicksort function to compare two records on a subset of fields.
-// Returns true if a is less than b.
-static bool _record_islt(Record a, Record b, const OpSort *op) {
-	return _record_compare(a, b, op) >
-		   0; // Return true if the current left element is less than the right.
+	return 0;
 }
 
-// Compares two heap record nodes.
-static int _heap_elem_compare(const void *A, const void *B, const void *udata) {
-	OpSort *op = (OpSort *)udata;
-	Record aRec = (Record)A;
-	Record bRec = (Record)B;
-	return _record_compare(aRec, bRec, op);
+static int _buffer_elem_cmp
```

Comment:
can we get rid of this func?

Reply:
We can't, `qsort` passes a pointers to the elements being compared, in this case `Record*` while our heap passes `Record` to its compare function.